storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/RemoteLogMetadataCache.java

```diff
@@ -125,29 +125,43 @@ public boolean isInitialized() {
      * @return the requested remote log segment metadata if it exists.
      */
     public Optional<RemoteLogSegmentMetadata> remoteLogSegmentMetadata(int leaderEpoch, long offset) {
-        RemoteLogLeaderEpochState remoteLogLeaderEpochState = leaderEpochEntries.get(leaderEpoch);
-
-        if (remoteLogLeaderEpochState == null) {
-            return Optional.empty();
+        RemoteLogSegmentMetadata metadata = getSegmentMetadata(leaderEpoch, offset);
```

Comment:
For my curiosity, where are we filtering for copy_segment_finished segments?

Reply:
In `copy_started` phase, we add the segmentId to [unreferenced-segment-ids](https://sourcegraph.com/github.com/apache/kafka/-/blob/storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/RemoteLogLeaderEpochState.java?L98). Once the copy completes, then we move that segmentId from unreference-segment-ids to [offsetToId](https://sourcegraph.com/github.com/apache/kafka/-/blob/storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/RemoteLogLeaderEpochState.java?L121) map, so the results returned from the RemoteLogLeaderEpochState is for the segments that are already in the `copy_finished` phase.

Reply:
Got it, thank you for explaining.